rfcs/0000-python-sdk.md

```diff
@@ -0,0 +1,276 @@
+
+```python
+driver_importer = importer.from_csv('driver_features.csv', granularity=granularity.DAY, entity='driver', owner='owner@email.com')
+```
+
+### [Alternative] Create an importer from a BigQuery table
+
+```python
+driver_importer_from_bq = importer.from_bq('project.dataset.drivers_table', granularity=granularity.DAY, entity='driver', owner='owner@email.com')
+```
+
+### [Alternative] Create an importer from a Pandas dataframe
+
+```python
+my_pandas_df = pd.read_csv('driver_features.csv')
+driver_importer_from_df = importer.from_df(my_pandas_df, granularity=granularity.DAY, entity='driver', owner='owner@email.com')
+```
+
+### Preview the dataframe loaded by the importer
+The `head` method prints out the first 5 rows of the Pandas dataframe.
+
+```python
+driver_importer.df.head()
+```
+
+### Change the local file path in the importer to GCS
+In order to ingest data from a bucket, it is necessary to modify the importer's path configuration to the location of the file that needs to be imported.
+
+```python
+driver_importer.options.path = 'gs://my-bucket-location/driver_features.csv'
+```
+
+### Describe the importer
+
+```python
+driver_importer.describe()
+```
+
+```yml
+type: file
+options:
+  format: csv
+  path: gs://my-bucket-location/driver_features.csv
+entities:
+  - driver
+schema:
+  entityIdColumn: driver_id
+  timestampColumn: ts
+  fields:
+    - name: timestamp
+    - name: customer_id
+    - name: completed
+      featureId: completed
+    - name: avg_distance_completed
+      featureId: avg_distance_completed
+    - name: avg_customer_distance_completed
+      featureId: avg_customer_distance_completed
+    - name: avg_distance_cancelled
+      featureId: avg_distance_cancelled
+```
+
+### Submit the import job
+The import job loads the CSV from GCS into Feast. It automatically registers entities and features with Feast during submission.
+
+```python
+fs.apply(driver_importer, create_entity=True, create_features=True)
+```
+_starting import..._  
+_10%_   
+_50%_   
+_100%_    
+_10 rows imported successfully_ 
+
+### Write out specification files for later use
+
+```python
+driver_importer.dump("driver_feature_import.yaml")
+customer_entity.dump("customer_entity.yaml")
+customer_age.dump("customer_entity.yaml")
+```
+
+### Create a “feature set” which can be used to query both training data and serving data.
+The feature set is simply an object that locally tracks which entity, granularity, and features you are interested in.
+
+```python
+feature_set = fs.create_feature_set(entity='driver', granularity='minute', features=['latitude', 'longitude', 'event_time'])
+```
+
+### Produce training dataset
+
+```python
+dataset_info = feature_set.create_training_dataset(start_date='2018-01-01', end_date='2018-02-01')
+```
+
+### Retrieve training dataset
+
+```python
+file_path = 'mypath.feather'
+dataset_info.download(destination=file_path, type='feather')
```

Comment:
Bigquery supports exporting into csv or avro but not to feather format. I can do conversion in sdk to convert csv/avro to feather, but that means the file has to be imported into memory and exported as feather (it seems defeat the purpose of downloading as a file). Do you think there is better approach? Or should we support download as csv/avro instead of feather? @woop

Reply:
@pradithya Good point. Internally we use feather more, but as a start I think it would make sense to support either of the 3 formats that BQ supports. CSV, JSON, or Avro. My preference for a first cut is Avro, since it retains type information.